dds/DCPS/Service_Participant.cpp

```diff
@@ -1606,146 +1606,109 @@ Service_Participant::load_configuration(
-                       domain_name.c_str(), repoKey.c_str()));
-          }
-        } else if (name == "DiscoveryConfig") {
-           repoKey = it->second;
-        } else if (name == "DefaultTransportConfig") {
-          if (it->second == "$file") {
-            // When the special string of "$file" is used, substitute the file name
-            perDomainDefaultTportConfig = ACE_TEXT_ALWAYS_CHAR(filename);
-
-          } else {
-            perDomainDefaultTportConfig = it->second;
-          }
+String
+Service_Participant::DomainConfig::default_transport_config(RcHandle<ConfigStoreImpl> config_store) const
+{
+  return config_store->get(config_key("DEFAULT_TRANSPORT_CONFIG").c_str(), "");
+}
 
-        } else {
-          ACE_ERROR_RETURN((LM_ERROR,
-                            ACE_TEXT("(%P|%t) Service_Participant::load_domain_configuration(): ")
-                            ACE_TEXT("Unexpected entry (%C) in [domain/%C] section.\n"),
-                            name.c_str(), domain_name.c_str()),
-                           -1);
-        }
-      }
+int
+Service_Participant::load_domain_configuration()
+{
+  const DCPS::ConfigStoreImpl::StringList sections = config_store_->get_section_names("OPENDDS_DOMAIN");
+  for (DCPS::ConfigStoreImpl::StringList::const_iterator pos = sections.begin(), limit = sections.end();
+       pos != limit; ++pos) {
+    const DomainConfig dc(*pos);
+    if (!process_domain(dc.to_domain(config_store_))) {
+      return -1;
+    }
+  }
 
-      if (domainId == -1) {
-        // DomainId parameter is not set, try using the domain name as an ID
-        if (!convertToInteger(domain_name, domainId)) {
-          ACE_ERROR_RETURN((LM_ERROR,
-                            ACE_TEXT("(%P|%t) Service_Participant::load_domain_configuration(): ")
-                            ACE_TEXT("Missing DomainId value in [domain/%C] section.\n"),
-                            domain_name.c_str()),
-                           -1);
-        }
-      }
+  return 0;
+}
 
-      if (!perDomainDefaultTportConfig.empty()) {
-        TransportRegistry* const reg = TransportRegistry::instance();
-        TransportConfig_rch tc = reg->get_config(perDomainDefaultTportConfig);
-        if (tc.is_nil()) {
-          ACE_ERROR_RETURN((LM_ERROR,
-            ACE_TEXT("(%P|%t) Service_Participant::load_domain_configuration(): ")
-            ACE_TEXT("Unknown transport config %C in [domain/%C] section.\n"),
-            perDomainDefaultTportConfig.c_str(), domain_name.c_str()), -1);
-        } else {
-          reg->domain_default_config(domainId, tc);
-        }
+bool
+Service_Participant::process_domain(const Domain& domain)
+{
+  if (domain.domain_id() == -1) {
+    // DomainId parameter is not set.
+    if (log_level >= LogLevel::Error) {
+      ACE_ERROR((LM_ERROR,
+                 "(%P|%t) Service_Participant::process_domain: "
+                 "Missing DomainId value in [domain/%C] section.\n",
+                 domain.name().c_str()));
```

Comment:
```suggestion
      ACE_ERROR((LM_ERROR,
                 "(%P|%t) ERROR: Service_Participant::process_domain: "
                 "Missing DomainId value in [domain/%C] section.\n",
                 domain.name().c_str()));
```
Some other logs also miss this part.